### ISSUE TITLE
Haversine Formula for Locations

### ISSUE THREAD

**Issue by Jacki3** [LoGaCulture/LUTE#56]
**Situation**
Currently, we determine if a player is at a specific location by using a radius (i.e., distance between the location and the player consistent with a set radius). Whilst this does work as expected, the accuracy and the overall sophistication of this process is lacking. Ideally, we need to take into account the curvature of the surface between two points and have a better way to define 'radius' by offering various units of measurement (metres, feet etc.).

**Solution**
Implement the Haversine Formula to ensure accuracy of distance between two points and update the location distance logic to accomodate for more user-friendly values.

**Additional context**
https://stackoverflow.com/questions/27928/calculate-distance-between-two-latitude-longitude-points-haversine-formula 
https://stackoverflow.com/questions/28790088/how-to-check-if-your-latitude-and-longitude-are-inside-a-circle

**Comment by Jacki3** [comment]
Location Variable now uses this formula and other areas that determine location should be updated; linked to the specific branch that will be merged with this implementation.
